### PR TITLE
APIGOV-25443 Added support for setting redaction config on logEventBuilder

### PR DESCRIPTION
**Pull request by kasuvy**

### CODE REVIEW

File: pkg/transaction/definitions.go [hunk 1]
```diff
@@ -50,20 +50,22 @@ const (
...
+	EnvironmentName        string   `json:"environmentName,omitempty"`
+	EnvironmentID          string   `json:"environmentId"`
+	TenantID               string   `json:"tenantId"`
+	TrcbltPartitionID      string   `json:"trcbltPartitionId"`
+	Type                   string   `json:"type"`
+	TargetPath             string   `json:"targetPath,omitempty"`
+	ResourcePath           string   `json:"resourcePath,omitempty"`
+	TransactionEvent       *Event   `json:"transactionEvent,omitempty"`
+	TransactionSummary     *Summary `json:"transactionSummary,omitempty"`
+	UnRedactedTargetPath   string
```
Comment (vivekschauhan) [inline pkg/transaction/definitions.go:67]: Should add json tag((`json:"-"`) to not serialize these properties

Comment (kasuvy) [reply pkg/transaction/definitions.go]: Fixed it in another commit.

File: pkg/transaction/logeventbuilder.go [hunk 2]
```diff
@@ -300,11 +304,45 @@ func (b *transactionEventBuilder) SetProtocolDetail(protocolDetail interface{})
...
+	} else {
+		b.logEvent.TargetPath, b.err = b.redactionConfig.URIRedaction(b.logEvent.UnRedactedTargetPath)
+	}
+	if b.err != nil {
+		return nil, b.err
+	}
+
+	//Set Resource Path
+	if b.logEvent.UnRedactedResourcePath == "" {
+		return nil, errors.New("unredacted resource path property not set in the LogEvent details")
```
Comment (jcollins-axway) [inline pkg/transaction/logeventbuilder.go:335]: what if b.logEvent.UnRedactedResourcePath is set while b.logEvent.UnRedactedTargetPath is not, looks like b.logEvent.UnRedactedResourcePath would not be redacted

File: pkg/transaction/definitions.go [hunk 3]
```diff
@@ -113,10 +113,11 @@ type Runtime struct {
...
 type EntryPoint struct {
-	Type   string `json:"type,omitempty"`
-	Method string `json:"method,omitempty"`
-	Path   string `json:"path,omitempty"`
-	Host   string `json:"host,omitempty"`
+	Type           string `json:"type,omitempty"`
+	Method         string `json:"method,omitempty"`
+	Path           string `json:"path,omitempty"`
+	Host           string `json:"host,omitempty"`
+	UnRedactedPath string `json:"-"`
```
Comment (jcollins-axway) [inline pkg/transaction/definitions.go:120]: Could you make the same change we did for the targetpath and resourcepath here, removing the UnRedacted* vars?

**Review by jcollins-axway — CHANGES_REQUESTED** [review]
comment above

File: pkg/transaction/definitions.go (hunk 3 above)
Comment (kasuvy) [reply pkg/transaction/definitions.go]: Added the change.

**Review by jcollins-axway — APPROVED** [review]

**Review by sbolosan — APPROVED** [review]

**Pull request merged**